File: apiclient/types/modelprovider.go
package types

type CommonProviderMetadata struct {
	Icon        string `json:"icon,omitempty"`
	IconDark    string `json:"iconDark,omitempty"`
	Description string `json:"description,omitempty"`
	Link        string `json:"link,omitempty"`
}

type CommonProviderStatus struct {
	CommonProviderMetadata
	Configured                      bool                             `json:"configured"`
	RequiredConfigurationParameters []ProviderConfigurationParameter `json:"requiredConfigurationParameters,omitempty"`
	OptionalConfigurationParameters []ProviderConfigurationParameter `json:"optionalConfigurationParameters,omitempty"`
	MissingConfigurationParameters  []string                         `json:"missingConfigurationParameters,omitempty"`
	Error                           string                           `json:"error,omitempty"`
}

type ProviderConfigurationParameter struct {
	Name         string `json:"name"`
	FriendlyName string `json:"friendlyName,omitempty"`
	Description  string `json:"description,omitempty"`
	Sensitive    bool   `json:"sensitive,omitempty"`
	Hidden       bool   `json:"hidden,omitempty"`
}

type ModelProvider struct {
	Metadata
	ModelProviderManifest
	ModelProviderStatus
}

type ModelProviderManifest struct {
	Name          string `json:"name"`
	ToolReference string `json:"toolReference"`
}

type ModelProviderStatus struct {
	CommonProviderStatus
	ModelsBackPopulated *bool `json:"modelsBackPopulated,omitempty"`
}

type ModelProviderList List[ModelProvider]

File: ui/admin/test/mocks/models/modelProvider.ts
import { ModelProvider } from "~/lib/model/providers";

export const mockedModelProvider: ModelProvider = {
	id: "openai-model-provider",
	created: "2025-02-04T16:04:02-05:00",
	revision: "1",
	type: "modelprovider",
	name: "OpenAI",
	toolReference: "github.com/obot-platform/tools/openai-model-provider",
	icon: "open-ai-logo-duotone.svg",
	link: "https://openai.com/",
	configured: true,
	requiredConfigurationParameters: [
		{
			name: "OBOT_OPENAI_MODEL_PROVIDER_API_KEY",
			friendlyName: "API Key",
			description:
				"OpenAI API Key. Can be created and fetched from https://platform.openai.com/settings/organization/api-keys or https://platform.openai.com/api-keys",
			sensitive: true,
		},
	],
	missingConfigurationParameters: ["OBOT_OPENAI_MODEL_PROVIDER_API_KEY"],
};
